.github/workflows/static_analysis.yml

```diff
@@ -0,0 +1,52 @@
+# contains tests that can run without compiling LightGBM, like
+# linters and documentation checks.
+name: Static Analysis
```

Comment:
Maybe add a badge for this new workflow?
https://github.com/microsoft/LightGBM/blame/master/README.md#L6

Reply:
good idea, added!